modules/dnn/test/test_layers_1d.cpp

```diff
@@ -245,4 +245,48 @@ INSTANTIATE_TEST_CASE_P(/*nothing*/, Layer_Elemwise_1d_Test, Combine(
 /*operation*/           Values("div", "prod", "max", "min", "sum")
 ));
 
+// WORKING FOR 1D
+typedef testing::TestWithParam<tuple<int>> Layer_Permute_Test;
+TEST_P(Layer_Permute_Test, Accuracy)
+{
+    LayerParams lp;
+    lp.type = "Permute";
+    lp.name = "PermuteLayer";
+
+    int order[] = {0}; // Since it's a 0D tensor, the order remains [0]
+    lp.set("order", DictValue::arrayInt(order, 1));
+
+    int dims = get<0>(GetParam());
+
+    Ptr<PermuteLayer> layer = PermuteLayer::create(lp);
+    std::vector<int> input_shape;
+    std::vector<int> output_shape;
+    Mat input;
+    if (dims > 1){
+        input_shape = {1, 10};
+        output_shape = {1, 10};
+        input = Mat(input_shape, CV_32F);
+    } else {
+        std::vector<int> input_shape = {dims};
```

Comment:
Why do we need `std::vector` here? Why regular array doesn't work like on the line 256?